### ISSUE TITLE
Incorrect resizing when opened by another terminal

### ISSUE THREAD

**Issue by rieje** [neovim/neovim#12993]
- `nvim --version`: v0.4.4
- `vim -u DEFAULTS` (version: ) behaves differently?
- Operating system/version: Arch Linux
- Terminal name/version: alacritty version 0.5.0
- `$TERM`: alacritty-direct

### Steps to reproduce:

On a tiling window manager like i3, open an instance of nvim with a terminal by calling: `<terminal> <--command> nvim`. In my case, `alacritty --command nvim`.

### Actual behaviour

You can see [here](https://imgbox.com/6EAb7IyI) that the statusline is in the middle of the window as opposed to the bottom of the window. If I focus on the window and then make it floating and then make it back to tiling, it will have the statusline in the correct position as shown [here](https://imgbox.com/qUjBqhd1).

### Expected behaviour

On vim, this is not an issue--the statusline is always at the bottom of the window.

**Comment by fidgetingbits** [comment]
- nvim --version: v0.5.0-828-g0a95549d6
- vim -u DEFAULTS (version: ) behaves the same
- Operating system/version: Arch Linux
- Terminal name/version: termite v15
- $TERM: xterm-256color

I am seeing this behavior on i3 window manager as well, I'm starting a terminal using the command `bindsym $mod+Return exec termite -e \"nvim -c ':term'\"`. This does not occur if the terminal does not run vim immediately. I can also confirm that other resizing events are correctly handled, so has noted above the workaround is to temporarily change the nvim window after opening and resizing it back, such is toggling i3's full screen mode. I have not tested any other tiling windows managers, so it is possible that that it could be an i3 bug.

I have tested this with the termite, alacritty, and kitty terminals. I've also confirmed that running vim with `-u NONE`, `-u DEFAULTS`, and my own config all behave the same.
